ipaserver/plugins/dogtag.py

```diff
@@ -1575,49 +1460,41 @@ def get_certificate(self, serial_number):
         """
         logger.debug('%s.get_certificate()', type(self).__name__)
 
-        # Convert serial number to integral type from string to properly handle
-        # radix issues. Note: the int object constructor will properly handle large
-        # magnitude integral values by returning a Python long type when necessary.
-        serial_number = int(serial_number, 0)
-
         # Call CMS
-        http_status, _http_headers, http_body = (
-            self._sslget('/ca/agent/ca/displayBySerial',
-                         self.env.ca_agent_port,
-                         serialNumber=str(serial_number),
-                         xml='true')
+        path = 'certs/{}'.format(serial_number)
+        _http_status, _http_headers, http_body = self._ssldo(
```

Comment:
Have you considered to replace ``_ssldo()`` with Dogtag's API ``pki.certs.CertClient.get_cert()``? I would prefer to slowly get rid of our own API wrapper and use pki client library instead.

Reply:
It's a fine idea.  Maybe for another commit?  I don't plan on spending more than the minimum time to get these patches working, reviewed and merged.  That means it's either take this as-is (subject to whatever minor changes are asked in review), or it will be abandoned.  I don't really mind either way - I just wanted to get all my old patches "out there" for consideration.

Reply:
SGTM! Let's get rid of ``_ssldo`` in another commit.

Reply:
Cool, thanks :)